src/tiledb/cloud/asset.py

```diff
@@ -11,30 +11,30 @@
 from .rest_api.models import GroupInfo  # type: ignore
 
 
-def delete(uri: str, recursive: bool = False) -> None:
+def delete_asset(uri: str, recursive: bool = False) -> None:
     """Deregister the asset and remove its physical groups and arrays from storage.
 
     :param uri: URI of the asset.
     :return: None.
     """
-    delete_map = {
+    delete_map: Mapping[str, Callable] = {
```

Comment:
I don't plan to add type hints to everything, but this is a key piece of data. Mypy, for example, will warn us if we misuse it.